contracts/InterchainTokenService.sol

```diff
@@ -1120,8 +1081,15 @@ contract InterchainTokenService is
      * @return address The token address.
      * @return uint256 The value the call is worth.
      */
-    function _contractCallValue(bytes calldata payload) internal view returns (address, uint256) {
-        (uint256 messageType, bytes32 tokenId, , , uint256 amount) = abi.decode(payload, (uint256, bytes32, bytes, bytes, uint256));
+    function _contractCallValue(bytes memory payload) internal view returns (address, uint256) {
+        uint256 messageType;
+        (messageType, , payload) = abi.decode(payload, (uint256, string, bytes));
```

Comment:
reuse `_getExecuteParams` to avoid duplication. Gas cost isn't an issue since this only gets queried by the express relayer